addons/survey/models/survey_user_input.py

```diff
@@ -149,6 +154,30 @@ def _compute_attempts_number(self):
 
                 user_input.attempts_number = attempts_number
 
+            self.env.cr.execute("""
```

Comment:
I feel like this SQL queries can be merged with the previous one (same group by and left outer join).

All the conditions are the same, maybe use a CTE (`with ... AS (SELECT ...)`) that will represent `all_attempts_user_input` and then deduce `previous_user_input` from it. It will remove redundancy of the conditions, make the code more elegant and speed up the execution as well (as the previous query is a subset of this one)

What do you think ?

Reply:
He be.